### PR TITLE
Error branch

### PR DESCRIPTION
**Pull request by lixingwang**
**What kind of change does this PR introduce?** (check one with "x")
```
[*] Bugfix
[] Feature
[] Code style update (formatting, local variables)
[] Refactoring (no functional changes, no api changes)
[] Other... Please describe:
```

**Fixes**: #

**What is the current behavior?**
Today we are not adding error data to error branch scope
**What is the new behavior?**
Add error data to the error branch to make error details available in the error branch

### CODE REVIEW

**Comment by fm-tibco** [comment]
Not sure if this is the right way to do this, since this a adding a global error.  lets say you have two tasks running in "parallel" that have an error-branch, if they each add the error globally, one will override the other.  We'll have to think about the correct way to do this.

File: definition/resolve.go [hunk 1]
```diff
@@ -37,7 +37,7 @@ func (r *FlowResolver) Resolve(scope data.Scope, itemName, valueName string) (in
 	if !exists {
 		return nil, fmt.Errorf("failed to resolve flow attr: '%s', not found in flow", valueName)
 	}
-
+	fmt.Println(value)
```
Comment (fm-tibco) [inline definition/resolve.go]: remove print

Comment (mellistibco) [inline definition/resolve.go]: do we need this println?

File: instance/taskinst.go [hunk 2]
```diff
@@ -425,9 +425,22 @@ func (ti *TaskInst) GetSetting(name string) (value interface{}, exists bool) {
 }
 
 func (ti *TaskInst) appendErrorData(err error) {
+	//For Golabla handle only
```
Comment (fm-tibco) [inline instance/taskinst.go]: should be "global"

File: instance/taskinst.go [hunk 3]
```diff
@@ -425,9 +425,22 @@ func (ti *TaskInst) GetSetting(name string) (value interface{}, exists bool) {
...
+	ti.flowInst.attrs["_E."+ti.Task().Name()] = errObj
+	ti.flowInst.SetValue("_E", errObj)
 
-	errorObj := NewErrorObj(ti.taskID, err.Error())
+}
 
+func (ti *TaskInst) setTaskError(err error) {
+	//For Golabla handle only
+	errObj := ti.getErrorObject(err)
+	ti.flowInst.attrs["_E."+ti.Task().Name()] = errObj
```
Comment (fm-tibco) [inline instance/taskinst.go]: Should be task ID

File: definition/resolve.go (hunk 1 above)
Comment (lixingwang) [reply definition/resolve.go]: thanks removed

File: instance/taskinst.go (hunk 3 above)
Comment (lixingwang) [reply instance/taskinst.go]: @fm-tibco  Updated

**Pull request merged**